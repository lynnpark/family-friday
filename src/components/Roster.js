/** @jsx jsx **/

import React from "react";
import { css, jsx } from "@emotion/core";
import Button from "./Button";
import NewHire from "./NewHire";
import { resetEmployees } from "../functions/api";

export default function Roster(props) {
  const reset = () => {
    resetEmployees().then(res => props.setEmployees(res));
  };

  return (
    <div css={css`border: 1px solid #ccc; padding: 10px;`}>
      <h2>
        Who's joining the fam today? <span aria-label="family" role="img">ðŸ‘ª</span>
      </h2>
      {props.employees ? (
        <React.Fragment>
          <NewHire
            addEmployee={employee => props.setEmployees([...props.employees, employee])}
          />
          <Button
            onClick={reset}
            label={"Reset Employees"}
            title={"Reset Employees"}
          />
        </React.Fragment>
      ) : (
        "Loading Employees..."
      )}
      <div>
        {props.employees && props.employees.map(employee => employee.name).join(", ")}
      </div>
    </div>
  );
}
