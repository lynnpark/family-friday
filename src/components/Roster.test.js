import React from "react";
import { render } from "@testing-library/react";
import Roster from "./Roster";

const emps = [
  { name: "Happy" },
  { name: "Dopey" },
  { name: "Grumpy" },
  { name: "Sneezy" },
  { name: "Lynn" },
  { name: "Bashful" }
];

describe("Roster Component", () => {
  test("Matches snapshot", () => {
    const component = render(<Roster employees={emps} setEmployees={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
