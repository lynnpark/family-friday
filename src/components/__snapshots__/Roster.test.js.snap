// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster Component Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-o000xv-Roster"
      >
        <h2>
          Who's joining the fam today? 
          <span
            aria-label="family"
            role="img"
          >
            ðŸ‘ª
          </span>
        </h2>
        <div
          class="newHire"
        >
          <h3>
            Add a new hire!
          </h3>
          <form>
            <input
              aria-label="Name of New Hire"
              class="newHire-name css-mnh0sj-NewHire"
              name="name"
              placeholder="ex. Lynn"
              type="text"
              value=""
            />
            <button
              aria-label="Add New Hire"
              class="css-hlrf09-Button"
            >
              Add
            </button>
          </form>
        </div>
        <button
          aria-label="Reset Employees"
          class="css-hlrf09-Button"
        >
          Reset Employees
        </button>
        <div>
          Happy, Dopey, Grumpy, Sneezy, Lynn, Bashful
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-o000xv-Roster"
    >
      <h2>
        Who's joining the fam today? 
        <span
          aria-label="family"
          role="img"
        >
          ðŸ‘ª
        </span>
      </h2>
      <div
        class="newHire"
      >
        <h3>
          Add a new hire!
        </h3>
        <form>
          <input
            aria-label="Name of New Hire"
            class="newHire-name css-mnh0sj-NewHire"
            name="name"
            placeholder="ex. Lynn"
            type="text"
            value=""
          />
          <button
            aria-label="Add New Hire"
            class="css-hlrf09-Button"
          >
            Add
          </button>
        </form>
      </div>
      <button
        aria-label="Reset Employees"
        class="css-hlrf09-Button"
      >
        Reset Employees
      </button>
      <div>
        Happy, Dopey, Grumpy, Sneezy, Lynn, Bashful
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
